core/src/main/java/org/apache/accumulo/core/fate/user/UserFateStore.java

```
@@ -151,6 +157,95 @@ protected void create(FateId fateId, FateKey fateKey) {
+        scanner.setRange(getRow(fateId));
+        scanner.fetchColumn(TxColumnFamily.RESERVATION_COLUMN.getColumnFamily(),
+            TxColumnFamily.RESERVATION_COLUMN.getColumnQualifier());
+        FateReservation persistedRes = scanner.stream()
+            .filter(entry -> FateReservation.isFateReservation(entry.getValue().toString()))
+            .map(entry -> FateReservation.from(entry.getValue().toString())).findFirst()
+            .orElse(null);
+        if (persistedRes != null && persistedRes.equals(reservation)) {
+          return Optional.of(new FateTxStoreImpl(fateId, reservation));
+        }
+      } catch (TableNotFoundException e) {
+        throw new IllegalStateException(tableName + " not found!", e);
+      }
+    }
+    return Optional.empty();
+  }
+
+  @Override
+  public boolean isReserved(FateId fateId) {
+    try (Scanner scanner = context.createScanner(tableName, Authorizations.EMPTY)) {
+      scanner.setRange(getRow(fateId));
+      scanner.fetchColumn(TxColumnFamily.RESERVATION_COLUMN.getColumnFamily(),
+          TxColumnFamily.RESERVATION_COLUMN.getColumnQualifier());
+      return scanner.stream()
+          .map(entry -> FateReservation.isFateReservation(entry.getValue().toString())).findFirst()
+          .orElse(false);
+    } catch (TableNotFoundException e) {
+      throw new IllegalStateException(tableName + " not found!", e);
+    }
+  }
+
+  @Override
+  public Map<FateId,FateReservation> getActiveReservations() {
+    Map<FateId,FateReservation> activeReservations = new HashMap<>();
+
+    try (Scanner scanner = context.createScanner(tableName, Authorizations.EMPTY)) {
+      scanner.setRange(new Range());
+      scanner.fetchColumn(TxColumnFamily.RESERVATION_COLUMN.getColumnFamily(),
+          TxColumnFamily.RESERVATION_COLUMN.getColumnQualifier());
+      scanner.stream()
+          .filter(entry -> FateReservation.isFateReservation(entry.getValue().toString()))
```

If we had buggy code that was writing out a reservation that looked like `<lockid>:<uuid>:<uuid>` I think `isFateReservation` would return false for this (not completley sure, hard to follow the curly braces in the diff).  I think it will split the string on `:` and return false if does not have two fields.  Would be better to make this more strict and throw an exception when unexpected data like this is seen.

---

Addressed. Now, isFateReservation() throws an error if it's passed anything unnexpected (only expects a NOT_RESERVED value or a valid FateReservation serialization).